src/prefect/schedules.py

```diff
@@ -233,3 +233,65 @@ def __init__(self, start_date: datetime):
         super().__init__(
             start_date=start_date, interval=timedelta(days=1), end_date=start_date
         )
+
+
+class UnionSchedule(Schedule):
+    """
+    A schedule formed by combining multiple other schedules.
+
+    Both `start_date` and `end_date` are inferred as the min / max (resp.) of
+    all provided schedules.  Note that the schedules are not required to all
+    be from the same timezone.
+
+    Args:
```

Comment:
Tiny docstring note: could you specify that if users provide overlapping schedules, only unique dates will be used? For example, if the same interval schedule was provided twice, this schedule would only run once for each date. (I'm pretty sure that's what you test, below)